### PR TITLE
Extract lit template support to a separate module

### LINKED ISSUE #9038: Extract Lit template support into a separate module
Consider extracting Lit template related code into a separate module so that `flow-server`  doesn't have it anymore
(since it's not a part of core functionality).

### PR DESCRIPTION
**Pull request by denis-anisimov**
Fixes #9038

### CODE REVIEW

File: flow-lit-template/pom.xml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.vaadin</groupId>
+        <artifactId>flow-project</artifactId>
+        <version>5.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>flow-lit-template</artifactId>
+    <name>Flow iLit Templates Support</name>
```
Comment (joheriks) [inline flow-lit-template/pom.xml:12]: Typo

File: flow-lit-template/pom.xml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.core</artifactId>
+            <version>${osgi.core.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <!--  Connect Dependencies -->
```
Comment (joheriks) [inline flow-lit-template/pom.xml:73]: Comment seems irrelevant

**Review by joheriks — CHANGES_REQUESTED** [review]
Should `flow-lit-template` also be managed by `flow-bom`?

File: flow-lit-template/src/main/java/com/vaadin/flow/component/littemplate/LitTemplate.java [hunk 3]
```diff
@@ -22,6 +22,7 @@
 import com.vaadin.flow.component.Tag;
 import com.vaadin.flow.component.dependency.JsModule;
 import com.vaadin.flow.component.littemplate.LitTemplateParser.LitTemplateParserFactory;
+import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
```
Comment (pleku) [inline flow-lit-template/src/main/java/com/vaadin/flow/component/littemplate/LitTemplate.java]: How can this work - polymer template is in a module that should not be visible

**Review by pleku — CHANGES_REQUESTED** [review]
One comment and please do add the modules to /platform too

**Comment by denis-anisimov** [comment]
> One comment and please do add the modules to /platform too

Which `platform` ?

**Review by pleku — APPROVED** [review]

**Comment by pleku** [comment]
> > One comment and please do add the modules to /platform too
> 
> Which `platform` ?

There is only one platform... but you'll need to add the modules to https://github.com/vaadin/platform/blob/master/vaadin-core/pom.xml

**Review by joheriks — APPROVED** [review]
